instance/tests/models/test_instance.py

```diff
@@ -436,3 +577,96 @@ def update_status(self, provisioned=False, rebooting=False):
         with patch('instance.models.server.OpenStackServer.start'):
             instance.provision()
         self.assertEqual(mock_run_playbook.call_count, 1)
+
+    @patch_os_server
+    @patch('instance.models.server.openstack.create_server')
+    @patch('instance.models.server.OpenStackServer.update_status')
+    @patch('instance.models.server.OpenStackServer.sleep_until_status')
+    @patch('instance.models.server.OpenStackServer.reboot')
+    @patch('instance.models.instance.gandi.set_dns_record')
+    @patch('instance.models.instance.OpenEdXInstance.run_playbook')
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_databases')
+    def test_provision_with_external_databases(self, os_server_manager, mock_provision_databases, mock_run_playbook,
+                                               mock_set_dns_record, mock_server_reboot, mock_sleep_until_status,
+                                               mock_update_status, mock_openstack_create_server):
+        """
+        Run provisioning sequence, with external databases
+        """
+        mock_openstack_create_server.return_value.id = 'test-run-provisioning-server'
+        os_server_manager.add_fixture('test-run-provisioning-server', 'openstack/api_server_2_active.json')
+
+        instance = OpenEdXInstanceFactory(sub_domain='run.provisioning', ephemeral_databases=False)
+        instance.provision()
+        self.assertEqual(mock_provision_databases.call_count, 1)
+
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_mysql')
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_mongo')
+    def test_provision_external_databases(self, mock_provision_mysql, mock_provision_mongo):
+        """
+        Provision mysql and mongo databases
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.provision_external_databases()
+        self.assertEqual(mock_provision_mysql.call_count, 1)
+        self.assertEqual(mock_provision_mongo.call_count, 1)
+
+    def test_provision_external_mysql(self):
+        """
+        Provision external mysql database
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.provision_external_mysql()
```

Comment:
@omarkhan Cf my comment above about moving tests that require mysql and mongo to be installed to integration tests. The unit tests above that mock the actual calls to the DBs are fine to keep here - and you could also keep this one and the others below by mocking the actual calls to the mysql/mongo drivers.